### DISCUSSION TITLE
How to hook into the startup for one-off initialization？

### DISCUSSION THREAD

**Discussion by ghost** [solidjs/solid-start#1447] [Q&A]
How does SolidStart execute certain tasks on the server side upon startup, such as background scheduled tasks, similar to using an instrumentation file in NestJS for configuration?

**Comment by rolanday** [comment]
@unlinking , you might want to take a look at https://github.com/solidjs/solid-start/discussions/1164. I has the same question back in Dec, and sadly received no help after many attempts so I moved on (still rooting for solid on the sideline though).

**Comment by emgeekc** [comment]
Same question here -- where to import background tasks so they are not lazy loaded, somewhere that works in both dev and production.

**Comment by ryansolid** [comment]
Can you put the code in `entry-server.tsx`? Not really clear what the requirement is here. Middleware is the other option if you can wait until the first request. But something top level there will only be ran on the server when first started up. You could also have a top level variable and if it isn't set yet have it be grabbed in middleware and set so subsequent requests wouldn't need to grab it again. 

@rolanday I did respond eventually to #1164 but I realize it took a while (3 months). Most people were incapable of answering your question at the time, and those who were, were focused on building out the features, including figuring out the answer to the question.

**Comment by emgeekc** [comment]
Hi Ryan, thanks for your attention. When I put the code into `entry-server.tsx` it fires on first request only.

In my personal use-case, I'm experimenting with an app that contains a local cache using SQLite for an outside database. Populating this cache can take up to 15 seconds or so and I'd prefer it done on first startup -- so by the time the first request comes in, cache is already primed and ready. 

Obviously I could refactor this step into a separate process, but would prefer not to, to keep the system simple.

**Reply by ryansolid** [reply]
That's not what I would have expected. The `createHandler` call in entry-server actually registers the endpoint so I would have expected it to run immediately rather than on receiving a request. It's where all SolidStarts's code begins (other than middleware which gets registered separately). @nksaraf do you know anything about the timing of when server startup runs?

**Reply by emgeekc** [reply]
This can be tested by starting a new project, adding a `console.log` to `entry-server.tsx` and starting it up. The `console.log` will only fire when the first request comes in. 

I'll admit I haven't tried changing any Solid Start configuration, so maybe if I enabled a different rendering mode things would be different.

**Reply by bauerbrun0** [reply]
In dev mode it will run on first request. With the release build it will run immediately (similarly to SvelteKit's [`hooks.server.js`](https://stackoverflow.com/a/74151215)).

**Reply by Sleepful** [reply]
If possible at all: how to modify the behavior of dev mode such that it behaves as the release build regarding execution of `entry-server.tsx`?
